mvp/app/src/main/java/com/dwarvesv/mvp/view/list/ListPresenter.kt

```diff
@@ -0,0 +1,70 @@
+package com.dwarvesv.mvp.view.list
+
+
+import android.content.Context
+import com.dwarvesv.mvp.R
+import com.dwarvesv.mvp.data.model.User.CREATOR.getListUserDummy
+import com.dwarvesv.mvp.data.request.GetUsersRequest
+import com.dwarvesv.mvp.repository.UserRepository
+import com.dwarvesv.mvp.utils.isNetworkConnected
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
+class ListPresenter(val view: ListContract.View, private val userRepository: UserRepository, private var context: Context?) : ListContract.Presenter {
+
+    init {
+        view.presenter = this
+    }
+
+    override fun getListData(limit: Int, offset: Int, pullToRefresh: Boolean) {
+
+        if (offset == 0) {
+            if (!pullToRefresh) view.showRefreshing()
+        } else view.showLoadMore()
+
+        if (!isNetworkConnected(context)) {
+
+            if (offset == 0)
+                view.hideRefreshing()
+            else
+                view.hideLoadMore()
+
+            view.showAlertDialog(R.string.title_connection_lost, R.string.msg_no_internet_connection)
+            return
+        }
+
+        userRepository.getListData(GetUsersRequest("")).observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(
+                        {
+                            if (it.body() != null) {
+
+                                callHideIndicator(offset)
+
+                                it.body()?.let { it1 ->
```

Comment:
we should never use anonymous name with number like `it1`, `it2`. If more than one `it`, please provide a specify name to make the logic cleaner

Reply:
@nambv: agree, it will fix